src/panels/lovelace/cards/hui-grid-card.ts

```diff
@@ -87,10 +87,14 @@ class HuiGridCard extends HuiStackCard<GridCardConfig> {
           grid-column: 1 / 1;
         }
 
-        :host([square]) #root > *:first-child {
+        :host([square]) #root > *:not([hidden]) {
           grid-row: 1 / 1;
           grid-column: 1 / 1;
         }
+        :host([square]) #root > *:not([hidden]) ~ *:not([hidden]) {
```

Comment:
Can we add a comment for this one? We probably won't remember what the heck this does and why in a few months

Reply:
I was thinking that git blame would lead to this PR with the description, but a comment is probably a good idea too, yes.